script.js

```diff
@@ -2,11 +2,104 @@
 function setup() {
   const allEpisodes = getAllEpisodes();
   makePageForEpisodes(allEpisodes);
+  selectEpisode(allEpisodes);
 }
+// converting objects into an
+// episodes = Object.entries(getAllEpisodes())
+// console.log(episodes);
+function makePageForEpisodes(allEpisodes) {
+  allEpisodes.forEach((episode, index) => {
+    // Create the HTML elements needed for a card
+    const div = document.createElement("div");
+    const h4TagElem = document.createElement("h3");
+    h4TagElem.classList = "episode-info";
+    const img = document.createElement("img");
+    const pElem = document.createElement("p");
+    // the body element so that the newly created elements
+    // can be appended to it
+    const rootElem = document.getElementById("root");
 
-function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+    // Use innerHTML to add text to the anchor tag
+    h4TagElem.innerHTML =
+      `${episode.name}` +
+      `-S${episode.season.toString().padStart(2, "0")}
+        E${episode.number.toString().padStart(2, "0")}`;
+    pElem.innerHTML = episode.summary;
+    // Append the elements to the DOM
+    rootElem.appendChild(div);
```

Comment:
For performance reasons it's generally better to build up your elements first, and only then append everything to the DOM at the end.

So you could create the rootElem here with all the cards inside it, and then attach it to your html body in one go. This would result in one repaint instead of 10s or 100s.

For example, check sections 1 and 2: https://howchoo.com/g/mmu0nguznjg/learn-the-slow-and-fast-way-to-append-elements-to-the-dom